index.js

```diff
@@ -3656,6 +3661,42 @@
       [Fn(a, b), m(a), m(a)],
       Z.sortBy);
 
+  //# zipWith :: (Applicative m, Foldable m, Monoid m) => (a -> b -> c) -> m a -> m b -> Arrray c
+  //.
+  //. Creates a new array out of the two supplied structures
+  //. by applying the given function to each equally-positioned
+  //. pair of elements from each given [Foldable][]. The returning
+  //. array is truncated to the length of the shorter of the two
+  //. input Foldables.
+  //.
+  //. ```javascript
+  //. > S.zipWith(S.add, [1, 2, 3], [3, 2, 1])
+  //. [4, 4, 4]
+  //.
+  //. > S.zipWith(a => b => [a, b], [1, 3, 5], [2, 4, 6])
+  //. [[1, 2], [3, 4], [5, 6]]
+  //.
+  //. > S.zipWith(a => b => Cons(a, Cons(b, Nil)), Cons('foo', Nil), Cons('bar', Nil))
+  //. [Cons('foo', Cons('bar', Nil))]
+  //. ```
+  function zipWith(f, a, b) {
+    var rv = [];
+    var idx = 0;
+    var as = toArray(a);
+    var bs = toArray(b);
+    var len = Math.min(as.length, bs.length);
+    while (idx < len) {
+      rv[idx] = f(as[idx])(bs[idx]);
+      idx += 1;
+    }
```

Comment:
I don't like having `idx` declared so far from its place of use. I would prefer a `for` loop:

```javascript
for (var idx = 0, min = Math.min(as.length, bs.length); idx < min; idx += 1) {
  ...
}
```

To appease the linter we may need to move one of the declarations:

```javascript
var min = Math.min(as.length, bs.length);
for (var idx = 0; idx < min; idx += 1) {
  ...
}
```